### PR TITLE
Add NetSuite REST adaptor

### LINKED ISSUE #103: NetSuite REST API Adaptor
Add support for the NetSuite REST API: https://docs.oracle.com/cloud/latest/netsuitecs_gs/NSTRW/NSTRW.pdf

Note:  "REST WEB SERVICES (BETA)" must be enabled on the account.

### PR DESCRIPTION
**Pull request by ryanwjackson**
Fix #103

### CODE REVIEW

**Comment by ryanwjackson** [comment]
@mattgmarcus @jozefvaclavik I'd love to get your review here!  I got full crud for customer working.  You can test it out using `qa.rb` if you want.

File: lib/ledger_sync.rb [hunk 1]
```diff
@@ -38,6 +39,8 @@
 require 'ledger_sync/result'
 require 'ledger_sync/adaptors/operation'
 require 'ledger_sync/adaptors/contract'
+require 'ledger_sync/adaptors/response'
```
Comment (ryanwjackson) [inline lib/ledger_sync.rb]: @jozefvaclavik assuming you like this pattern, it'd be nice to wrap our other requests in these objects for consistency.

File: lib/ledger_sync/adaptors/ledger_serializer_attribute_set.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def add(attribute)
         end
 
         if attribute.resource_attribute.present?
-          raise "resource_attribute already defined for #{serializer_class.name}: #{attribute.resource_attribute}" if resource_attribute_keyed_hash.key?(attribute.resource_attribute.to_s)
+          # raise "resource_attribute already defined for #{serializer_class.name}: #{attribute.resource_attribute}" if resource_attribute_keyed_hash.key?(attribute.resource_attribute.to_s)
```
Comment (ryanwjackson) [inline lib/ledger_sync/adaptors/ledger_serializer_attribute_set.rb]: I had to remove this line, as expected.  This PR also largely ignores that we need to split the serializers up for this adaptor.  We will need a serializer and deserializer, which are both essentially intelligent "mergers."

File: lib/ledger_sync/adaptors/netsuite_rest/token.rb [hunk 3]
```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+# ref: https://docs.oracle.com/cloud/latest/netsuitecs_gs/NSATH/NSATH.pdf
+# ref: Search "The Signature for Web Services and RESTlets" in NetSuite documentation (must have an account)
+module LedgerSync
+  module Adaptors
+    module NetSuiteREST
+      class Token
+        DEFAULT_SIGNATURE_METHOD = 'HMAC-SHA256'
```
Comment (mattgmarcus) [inline lib/ledger_sync/adaptors/netsuite_rest/token.rb]: This is the one that works I'm assuming?

Comment (ryanwjackson) [reply lib/ledger_sync/adaptors/netsuite_rest/token.rb]: Yes.  I believe both work `HMAC-SHA256` and `HMAC-SHA1`

**Pull request merged**